src/parser/jobs/ast/modules/ArcanaTracking/ArcanaSuggestions.tsx

```diff
@@ -56,172 +59,160 @@ interface SleeveIcon {
+	targetJob: string | null
 }
 
-export default class ArcanaSuggestions extends Module {
+export default class ArcanaSuggestions extends Analyser {
 	static override handle = 'arcanaSuggestions'
 
 	static override title = t('ast.arcana-suggestions.title')`Arcana Logs`
 	static override displayOrder = DISPLAY_ORDER.ARCANA_TRACKING
 
 	@dependency private data!: Data
-	@dependency private combatants!: Combatants
 	@dependency private arcanaTracking!: ArcanaTracking
 	@dependency private timeline!: Timeline
+	@dependency private actors!: Actors
 
 	private cardLogs: CardLog[] = []
-	private partyComp: string[] = []
-
-	private PLAY: number[] = []
-
-	private SLEEVE_ICON: SleeveIcon = {}
+	private play: Array<Action['id']> = []
+	private sleeveIcon: SleeveIcon = {}
 
 	//div privates for output
-	private history: DIVWindow[] = []
-	private lastDIVFalloffTime: number = 0
-	private divCast: number = 0
+	private history: DivinationWindow[] = []
+	private divinationCast: number = 0
 	//end div privates
 
-	protected override init() {
-		PLAY.forEach(actionKey => {
-			this.PLAY.push(this.data.actions[actionKey].id)
-		})
+	override initialise() {
+		this.play = PLAY.map(actionKey => this.data.actions[actionKey].id)
 
-		this.SLEEVE_ICON = {
+		this.sleeveIcon = {
 			[SleeveType.NOTHING]: '',
 			[SleeveType.ONE_STACK]: this.data.statuses.SLEEVE_DRAW.icon,
 			[SleeveType.TWO_STACK]: 'https://xivapi.com/i/019000/019562.png',
 		}
 
 		//used for divination tracking
-		this.addEventHook('normalisedapplybuff', {to: 'player', abilityId: this.data.statuses.DIVINATION.id}, this.tryOpenWindow)
-		this.addEventHook('normalisedapplybuff', {by: 'player', abilityId: this.data.statuses.DIVINATION.id}, this.countDIVBuffs)
-		this.addEventHook('removebuff', {to: 'player', abilityId: this.data.statuses.DIVINATION.id}, this.tryCloseWindow)
-		this.addEventHook('cast', {by: 'player'}, this.onCast)
+		const divinationFilter = filter<Event>().type('statusApply').status(this.data.statuses.DIVINATION.id)
+		this.addEventHook(divinationFilter.target(this.parser.actor.id), this.tryOpenWindow)
+		this.addEventHook(divinationFilter.source(this.parser.actor.id), this.countDivinationBuffs)
+		this.addEventHook(
+			filter<Event>()
+				.type('statusRemove')
+				.target(this.parser.actor.id)
+				.status(this.data.statuses.DIVINATION.id),
+			this.tryCloseWindow,
+		)
+		this.addEventHook(filter<Event>().type('action').source(this.parser.actor.id), this.onCast)
 
 		this.addEventHook('complete', this._onComplete)
 	}
 
 	//functions for divination buff counting
-	private countDIVBuffs(event: NormalisedApplyBuffEvent) {
+	private countDivinationBuffs(event: Events['statusApply']) {
 		// Get this from tryOpenWindow. If a window wasn't open, we'll open one.
-		const lastWindow: DIVWindow | undefined = this.tryOpenWindow(event)
+		// If it was already open (because another Astrologian went first), we'll keep using it
+		const lastWindow: DivinationWindow = this.tryOpenWindow(event)
```

Comment:
I am unsure how to implement this

Reply:
Have a new var like `private currentWindow: DivinationWindow | undefined`, then store stuff in that and use a null check to see if it's active. In the callback for where the window closes, you push the currentWindow to history and then set it to undefined. That way you know what's active separately to the history, don't need to iterate the array, and know that your full window is in history without any corruption. Just remember to close the window in onComplete in case it's open when the fight ends. Most jobs with a window of some sort do this so plenty of examples.

Reply:
works beautifully! thanks!